esp/services/ws_fs/ws_fsService.cpp

```diff
@@ -3261,39 +3169,32 @@ bool CFileSprayEx::onDropZoneFiles(IEspContext &context, IEspDropZoneFilesReques
         if (dropZoneList.ordinality())
             resp.setDropZones(dropZoneList);
 
-        if (netAddressStr.length() < 1)
-            return true;
-
-        char pathSep = '/';
-        if (osStr && *osStr)
-        {
-            int os = atoi(osStr);
-            if (os == OS_WINDOWS)
-                pathSep = '\\';
-        }
+        const char* dzName = req.getDropZoneName();
+        const char* directory = req.getPath();
+        const char* subfolder = req.getSubfolder();
 
-        directoryStr.replace(pathSep=='\\'?'/':'\\', pathSep);
+        if (isEmptyString(netAddress) || (isEmptyString(directory) && isEmptyString(subfolder)))
+            return true;
 
-        if (subfolder && *subfolder)
+        StringBuffer netAddressStr, directoryStr, osStr;
+        netAddressStr.set(netAddress);
+        if (!isEmptyString(directory))
+            directoryStr.set(directory);
+        if (!isEmptyString(subfolder))
         {
-            if (*(directoryStr.str() + directoryStr.length() -1) != pathSep)
-                directoryStr.append( pathSep );
-
+            if (directoryStr.length())
+                addPathSepChar(directoryStr);
             directoryStr.append(subfolder);
         }
+        addPathSepChar(directoryStr);
 
-        if (*(directoryStr.str() + directoryStr.length() -1) != pathSep)
-            directoryStr.append( pathSep );
+        getDropZoneFiles(context, dzName, netAddress, directoryStr.str(), req, resp);
 
-        PROGLOG("getDropZoneFiles: netAddress %s, path %s", netAddressStr.str(), directoryStr.str());
-        getDropZoneFiles(context, netAddressStr.str(), osStr.str(), directoryStr.str(), req, resp);
-
-        if (pathSep=='\\')
-            directoryStr.replaceString("\\", "\\\\");
-
-        resp.setNetAddress(netAddressStr.str());
+        resp.setDropZoneName(dzName);
```

Comment:
What's confusing is that it returns all dropzones, and in response returns the dzName (if supplied).
Why does it need to return 1 dropzone explicitly + all ?